lib/rules/no-extra-boolean-cast.js

```diff
@@ -63,6 +63,53 @@ module.exports = function(context) {
                     grandparent.callee.name === "Boolean")) {
                 context.report(node, "Redundant double negation in Boolean constructor call.");
             }
+        },
+        "CallExpression": function(node) {
+            var parent = node.parent;
+
+            if (node.callee.type !== "Identifier" || node.callee.name !== "Boolean") {
+                return;
+            }
+
+            // if (<bool>) ...
+            if (parent.type === "IfStatement") {
+                context.report(node, "Redundant Boolean call in an if statement condition.");
+
+            // do ... while (<bool>)
+            } else if (parent.type === "DoWhileStatement") {
+                context.report(node, "Redundant Boolean call in a do while loop condition.");
+
+            // while (<bool>) ...
+            } else if (parent.type === "WhileStatement") {
+                context.report(node, "Redundant Boolean call in a while loop condition.");
+
+            // <bool> ? ... : ...
+            } else if ((parent.type === "ConditionalExpression" &&
+                    node === parent.test)) {
+                context.report(node, "Redundant Boolean call in a ternary condition.");
+
+            // for (...; <bool>; ...) ...
+            } else if ((parent.type === "ForStatement" &&
+                    node === parent.test)) {
+                context.report(node, "Redundant Boolean call in a for loop condition.");
+
+            // !<bool>
+            } else if ((parent.type === "UnaryExpression" &&
+                    parent.operator === "!")) {
+                context.report(node, "Redundant Boolean call.");
+
+            // Boolean(<bool>)
```

Comment:
Included the next two checks to match `!!`, but they may be unnecessary; `Boolean(Boolean(...))` seems unlikely.